backend/interf.ml

```diff
@@ -79,11 +81,13 @@ let build_graph fundecl =
      do not add an interference between them if the source is still live
      afterwards. *)
   let add_interf_move src dst s =
+    if debug then assert (Reg.types_are_compatible src dst);
```

Comment:
I would be tempted to have this assertion even when `debug` is `false`. It is probably cheap, and would catch more cases when a move between incompatible types has been produced. ("more" meaning `emit.mlp` would catch moves present until the end while here we would also catch bad intentions.)

Reply:
Please also use `Misc.fatal_errorf` so when an assertion fails, we don't necessarily need to rebuild to get useful debugging info.

Reply:
(Sorry, I meant to attach this comment to the `add_pref` instance, where it makes more sense.)

Reply:
Sure, enabled the assertions unconditionally (except the collision check)

Reply:
Oh, didn't see the new comments - updated with error messages